gdpr/impl.go

```diff
@@ -61,18 +72,51 @@ func (p *permissionsImpl) AuctionActivitiesAllowed(ctx context.Context, bidderCo
 		return DenyAll, nil
 	}
 
-	weakVendorEnforcement := p.cfg.BasicEnforcementVendor(bidder)
+	// note the bidder here is guaranteed to be enabled
+	vendorID, vendorFound := p.resolveVendorID(bidderCoreName, bidder)
+	basicEnforcementVendors := p.cfg.BasicEnforcementVendors()
+	_, weakVendorEnforcement := basicEnforcementVendors[string(bidder)]
+
+	if !vendorFound && !weakVendorEnforcement {
+		return DenyAll, nil
+	}
+
+	parsedConsent, vendor, err := p.parseVendor(ctx, vendorID, p.consent)
+	if err != nil {
+		return DenyAll, err
```

Comment:
This might be out of the scope of this PR, but I know you moved some stuff around for code cleanliness, but this line isn't covered by tests, could it be updated?

Reply:
Ah yes we haven't had good test coverage in `impl.go` for a long time. I'm aware of this. I was planning to address the test coverage holes in `impl.go` in a future PR. Also a lot of the tests there will be rewritten. I wanted to keep those tests as is for now to increase confidence that my refactors aren't breaking something.